### DISCUSSION TITLE
Welcome to Print-To-3D Discussions!

### DISCUSSION THREAD

**Discussion by p3d-dev** [p3d-dev/Print-To-3D#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jmger** [comment]
Hi I usually use CURA 5.x. I like CURA but I am looking for an iPad version. I just bought the M2 iPad Pro. Using touch for me is like using Wireless vs wired Ear Phones. It is just so much free'er (spelling). I use Shapr3D on my iPad for the same reason. I have 3 printers. A Ender 3 Max that I highly modified to the point that you can't call it an Ender anymore. Using the New E3D REVO hot end. A modified Tevo Tornado. And a Monoprice Mini Delta printer that I need to repair. I am working on several designs but have not yet got a reliable enough printer setup in order to print my designs. One design has about 10 parts and each part takes around 17-20 hours to print. I have a Question: how do I rotate my part in Print To 3D? And how do I Auto level my part to the bed? Is there a user manual for Print To 3D?

Thank You.

**Reply by p3d-dev** [reply]
Thanks for the feedback and we believe similarly, that for working with 3D objects the touch UI of iPad/iPhone is way more intuitive than using a mouse on a traditional PC. Thereof, a slicer application should benefit of this ease of use, too. Just evolving with exactly the right interactions is not easy, so the part rotation is perhaps not as intuitive as it should be, Anyway, PrintTo3D is heading in this direction, but still we are only at the beginning and there is plenty of work to do. Currently we are reworking the UI gesture processing and may change some interactions. Until now the UI has evolved quickly and the constant update of user documentation would have wasted a lot of our development resources. But it is still on our todo list and first results should be available end of this/beginning of next year. Means: there is no user manual.

The key for object modification is to be able to select a print object. In earlier releases, this selection was broken for some iOS/iPad versions and hopefully this was fixed. If you tap an object on the print bed and the print object turns color from yellow to green, then that object should be selected. From the left (on iPad), a dock should slide in with six buttons.

The second button from top enables a mode, which should support leveling the part by touching any surface on the part, which then aligns this surface's normal with the print bed. In the past, there was an automatic leveling implemented, but that feature was too slow and so we decided to disable it until this feature has matured. For movement of the print object on the printbed, just tap and drag the object. Rotation around the z-axes can be done by a pan gesture (not starting on the print object). The object can be rotated by a two finger rotating gesture. If the object only rotates around the z-axes, then turn the printbed, so that the print object is viewed from the side. From this position, the object rotation should be around the view axes.

In case the print object cannot be selected and turns green, or any operation has a problem, please let us know, so we can fix it.

**Reply by emmanuel-a-otchere** [reply]
Hi I just downloaded the App, and I love this initiative and really would be looking forward to support it in the future. 
I've configured by Printer and Material profile, as well as connected it and tested the connection to my Octoprint. I loaded a test model, and i hit the slice button, am not sure what happens next, as the print button is disabled, as is the 4 icons on the far top-right of the landing UI. Any ideas what I'm missing?

**Reply by p3d-dev** [reply]
Thanks for contacting us. From your explanation the steps sound alright. To narrow down the issue could you please help with a screen recording  and sending us the video ? The video should show the printer/gcode settings of your configured printer, the process settings, loading an object and pressing the slice button. Sorry for this inconvenience, but it should enable us to help you.

**Comment by weslijones** [comment]
Can you provide a list of feature/services available for each subscription level?

Thanks

**Reply by p3d-dev** [reply]
Each subscription level offers the possibility to export the generated gcode either via Share or Printer upload.

The subscriptions are addressing the user needs:

students can enjoy gcode upload with the lowest yearly price as allowed by the AppStore
regular users can select between weekly, monthly and yearly plans - depending on the expected duration of their project(s)
enthusiasts - willing to support and push for rapid development - can do so by choosing the quarterly or weekly donation plans

**Reply by p3d-dev** [reply]
We have copied our comment from last year as direct reply.

**Comment by weslijones** [comment]
Can I use a subscription on two or three devices at the same time?Wesli JonesOn Nov 27, 2022, at 12:05 PM, p3d-dev ***@***.***> wrote:﻿
Each subscription level offers the possibility to export the generated gcode either via Share or Printer upload.
The subscriptions are addressing the user needs:

students can enjoy gcode upload with the lowest yearly price as allowed by the AppStore
regular users can select between weekly, monthly and yearly plans - depending on their project(s) expected duration
enthusiasts - willing to support and push for rapid development - can do so by choosing the quarterly or weekly donation plans

**Reply by p3d-dev** [reply]
This is controlled by Apple. From our understanding, you can share between your devices and with family members, too.

**Reply by p3d-dev** [reply]
We have copied our comment from last year as direct reply.

**Comment by ayman303350** [comment]
Dear Print to 3d dev 
Last version downloaded from app store with no share or print ?????.?

Where is the option to upgrade my copy

**Reply by p3d-dev** [reply]
Dear ayman303350,

sorry for the inconvenience. Based on your input, we have installed the latest version and afterwards our latest test version. Both times, we cannot reproduce this issue.

In the sidebar, there is a section "About". If you tap on this, you will be presented with a link to Twitter, GitHub-Feedback, the print to 3D version (latest is v3.9-2076) and info about purchases. In case that info shows "Refresh Purchases", then this is a button. Please tap it. The app will ask the AppStore for any valid subscription. If positive result is retrieved or the info could already be retrieved at app start automatically, then that info should display the known subscription with expiration date.

In case this does not help, please support us on narrowing down the problem:
- You are using an iPhone or an iPad ?
- Could you please provide a screenshot with a sliced object on the printbed and opened sidebar with the About section expanded. For an iPhone in addition: if the right buttons are combined into a submenu - reachable by a three dot icon, then please press that icon once, so the list of options can be seen in the screenshot.

**Comment by ayman303350** [comment]
![image](https://user-images.githubusercontent.com/104399062/205436632-6b916d7b-c816-48ee-9ea3-e99f05a661ac.jpeg)

Thanks for reply speed

**Reply by p3d-dev** [reply]
Thanks for the fast reply with the screenshot. This helps a lot. Apparently the model is not yet sliced. So please press the button left from the camera button: the staple of three squares. If the slicing has been successfully performed, then the currently grayed out buttons will be enabled. If this does not happen for you, please provide another screenshot for our next step in the investigation.

The purchase options are popping up after trying to Share or in the printer upload view on Start Upload. Eventually we will offer a more direct path to the purchase options via the About section.

**Reply by ayman303350** [reply]
![image](https://user-images.githubusercontent.com/104399062/205437282-47df8431-53e0-46b2-831c-92448c1ba0d9.jpeg)

Thanks 
Its ok now

**Reply by p3d-dev** [reply]
Happy to hear, that it works for you now. Enjoy Print To 3D, slicing at your fingertips.

**Comment by ayman303350** [comment]
Thanks a lot for the update

**Comment by weslijones** [comment]
Is there a way to export our process configurations settings? I’m trying to tweak my print quality. I need to document each change that I make so I know what works and what doesn’t work. Right now the only way you can do that is by taking a bunch of screenshots, it would be a lot easier if I could save them as text somewhere.

Also, I’m not sure what some of the process configurations settings do. Is there a place to get clear explanations? Currently when I tap the information circle, some settings have very brief explanations.

**Reply by p3d-dev** [reply]
A textual output in json-format would be quite easy and this would allow us to import previously exported settings. Would this be sufficient for you ?

Unfortunately, there is no more documentation for the slicer engine settings available. For which of the settings do you have further questions, so we can try to extend the information provided by the info button ?

**Comment by Dan-De-Sousa** [comment]
Any plans to add Klipper compatibility?

**Reply by p3d-dev** [reply]
As far as we understand it, Klipper drives the 3d printer and moonraker is used for the communication. PrintTo3D is compatible with moonraker for gcode upload. What is missing from your point of view?

**Reply by Dan-De-Sousa** [reply]
From what I understand, Klipper GCode is slightly different than the Marlin GCode. There are or at least were several commands not interchangeable.

From my experience after importing the GCode generated by the the app. The printer executes the Preamble with no issues, but then the code does not behave as expected.

**Reply by p3d-dev** [reply]
Judging by [klipper-docu](https://www.klipper3d.org/Slicers.html), marlin flavor should be chosen for the sliceramd should be compatible. Only the nifty klipper feature around motor control may lead to surprising results. Could you please be more specific, what the problem is, which you are observing ?

**Reply by Dan-De-Sousa** [reply]
While using the app, I haven't been able to reproduce the same issue twice yet. Now I'm not so sure it is Klipper related at all.

I have successfully printed several controlled tests with known settings slicing them with Cura and SuperSlicer. In attempting to transfer these settings to the app I may have missed some areas. So I guess I'm still having to learn your system. For example:
What do these mean?
Shell Factor:
Solid Factor:
Infill Factor:

![image](https://user-images.githubusercontent.com/35589893/229008020-c46e4449-e28c-498d-8f4f-a05b8a92172d.jpeg)

**Reply by p3d-dev** [reply]
Those are additional factors, which are multiplied to the extruded amount of filament. 

Example:
- slicing of 3DBenchy with 1.25 for all three factors yields a use of 5216mm filament
- slicing with factors 1.5 needs 6523mm filament
- slicing with factors 1.0 needs 3996mm filament

Calculating the factors from the filament usage yields: 6523/3996=1.63 and 5216/3996=1.3
Not an ideal match, but comes quite close
